nav2_dwb_controller/dwb_plugins/include/dwb_plugins/kinematic_parameters.hpp

```diff
@@ -113,13 +107,34 @@ class KinematicParameters
   double min_speed_xy_sq_{0};
   double max_speed_xy_sq_{0};
 
-  void reconfigureCB();
+  friend class KinematicsHandler;
+};
+
+/**
+ * @class KinematicsHandler
+ * @brief A class managing the representation of the robot's kinematics
+ */
+class KinematicsHandler
+{
+public:
+  KinematicsHandler();
+  ~KinematicsHandler();
+  void initialize(const nav2_util::LifecycleNode::SharedPtr & nh, const std::string & plugin_name);
+
+  inline KinematicParameters getKinematics() {return *kinematics_.load();}
+
+  bool isValidSpeed(double x, double y, double theta);
+
+  using Ptr = std::shared_ptr<KinematicsHandler>;
+
+protected:
+  std::atomic<KinematicParameters *> kinematics_;
```

Comment:
this can just be a variable, I don't think its nearly large enough to warrant a pointer

Reply:
The only reason why I ended up using a pointer here is that I was getting linking errors when trying to define the variable as atomic, because it is a custom class I believe. 
When I saw those errors I thought instantly in compiler and architecture problems and I thought about the pointer, which works well out of the box with the atomic class.
I agree that for this case a simple variable would be better, since we would get rid of the memory management. I'll look into the steps to use this as a variable and let you know if I find a solution and how problematic it is. Please let me know if you know more about it.

Reply:
Oh, I didn't know that. Google agrees. All good, don't worry about it, pointer is fine.